### PR TITLE
Helm install tests for ambient multi-namespace deployment scenarios

### PR DESCRIPTION
**Pull request by jmcshane**
**Please provide a description of this PR:**

Tests helm installation and verification of ztunnel and istio-cni charts in non `istio-system` namespace

Begins #49924

### CODE REVIEW

File: tests/integration/helm/util.go [hunk 1]
```diff
@@ -256,58 +311,63 @@ func CreateNamespace(t test.Failer, cs cluster.Cluster, namespace string) {
...
+	if err := h.DeleteChart(BaseReleaseName, config.IstioNamespace()); err != nil {
 		t.Errorf("failed to delete %s release: %v", BaseReleaseName, err)
 	}
-	if err := cs.Kube().CoreV1().Namespaces().Delete(context.TODO(), IstioNamespace, metav1.DeleteOptions{}); err != nil {
-		t.Errorf("failed to delete istio namespace: %v", err)
-	}
-	if err := kubetest.WaitForNamespaceDeletion(cs.Kube(), IstioNamespace, retry.Timeout(RetryTimeOut)); err != nil {
-		t.Errorf("waiting for istio namespace to be deleted: %v", err)
+	for _, ns := range config.Namespaces() {
+		if ns == "kube-system" {
```
Comment (MorrisLaw) [inline tests/integration/helm/util.go:334]: we can use the constant value here:
```go
if ns == constants.KubeSystemNamespace {
    continue
}
```

**Comment by jmcshane** [comment]
Refactored. This now uses the `NamespacedName` concept to map the helm release name to a target namespace. Additional helm install scenarios should be able to be added as needed

**Comment by jmcshane** [comment]
Will have to take a different approach for the `istioctl` install commands, but could likely use the name of top level components and generate override values to merge with the existing tests (that will be another PR)

File: tests/integration/helm/install_test.go [hunk 2]
```diff
@@ -40,18 +42,54 @@ global:
...
-		Run(setupInstallation(ambientProfileOverride, true))
+		Run(setupInstallation(ambientProfileOverride, true, DefaultNamespaceConfig))
+}
+
+func TestAmbientInstallMultiNamespace(t *testing.T) {
+	tests := []struct {
+		name     string
+		nsConfig NamespaceConfig
+	}{{
+		name: "istio-cni",
```
Comment (bleggett) [inline tests/integration/helm/install_test.go:61]:
```suggestion
		name: "isolated-istio-cni",
```

IIUC might be good to use a more descriptive case name here.

File: tests/integration/helm/install_test.go [hunk 3]
```diff
@@ -40,18 +42,54 @@ global:
...
+		nsConfig: NewNamespaceConfig(types.NamespacedName{
+			Name: CniReleaseName, Namespace: "istio-cni",
+		}),
+	}, {
+		name: "istio-cni-ztunnel",
```
Comment (bleggett) [inline tests/integration/helm/install_test.go:66]:
```suggestion
		name: "isolated-istio-cni-and-ztunnel",
```

File: tests/integration/helm/install_test.go [hunk 4]
```diff
@@ -40,18 +42,54 @@ global:
...
+		nsConfig: NewNamespaceConfig(types.NamespacedName{
+			Name: CniReleaseName, Namespace: "istio-cni",
+		}, types.NamespacedName{
+			Name: ZtunnelReleaseName, Namespace: "kube-system",
+		}),
+	}, {
+		name: "istio-cni-foo",
```
Comment (bleggett) [inline tests/integration/helm/install_test.go:73]:
```suggestion
		name: "isolated-istio-cni-ztunnel-and-gateway",
```

**Review by bleggett — APPROVED** [review]
LGTM, ty, few naming nits if I read it right.

**Comment by jmcshane** [comment]
@bleggett updated those more specific test names into a single commit. Should be ready for review once again

**Review by MorrisLaw — APPROVED** [review]

**Review by bleggett — APPROVED** [review]
LGTM!

**Review by dhawton — APPROVED** [review]

**Pull request merged**